managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -426,6 +454,85 @@ public void operationFailed(ManagedLedgerException exception) {
         });
     }
 
+    public void recoverPendingAckPosition(CompletableFuture<SubscriptionPendingAckMessages> future) {
+        // Read the meta-data ledgerId from the store
+        log.info("[{}] Recovering pending ack messages from bookkeeper ledger cursor: {}", ledger.getName(), name);
+        ledger.getStore().asyncGetSubscriptionPendingAckMessages(ledger.getName(), name, new MetaStoreCallback<SubscriptionPendingAckMessages>() {
+            @Override
+            public void operationComplete(SubscriptionPendingAckMessages subscriptionPendingAckMessages, Stat stat) {
+
+                pendingAckMessagesStoreStat = stat;
+
+                if (subscriptionPendingAckMessages.getPendingAckMsgLedgerId() == -1L) {
+                    // Pending ack messages info is stored in ledger meta store.
+                    future.complete(subscriptionPendingAckMessages);
+                    startCreateNewPendingAckMsgLedger();
+                    return;
+                } else {
+                    // Pending ack message info is stored in ledger.
+                    ledger.mbean.startCursorLedgerOpenOp();
+                    long ledgerId = subscriptionPendingAckMessages.getPendingAckMsgLedgerId();
+                    bookkeeper.asyncOpenLedger(ledgerId, digestType, config.getPassword(), (rc, lh, ctx) -> {
+                        if (isBkErrorNotRecoverable(rc)) {
+                            log.error("[{}] Cursor {} fail to open pending ack message ledger {} for recover",
+                                    ledger.getName(), name, ledgerId);
+                            startCreateNewPendingAckMsgLedger();
```

Comment:
If we fail to open the ledger, why do we call `startCreateNewPendingAckMsgLedger` here?